AppCenterData/AppCenterData/Internal/Client/MSDataOperationProxy.m

```diff
@@ -162,17 +164,74 @@ - (void)performOperation:(NSString *_Nullable)operation
-        [self updateLocalStore:token
-            currentCachedDocument:cachedDocument
-                newCachedDocument:documentWrapper
-                 deviceTimeToLive:deviceTimeToLive
-                        operation:operation];
+        [self.documentStore updateDocumentWithToken:token
+                              currentCachedDocument:cachedDocument
+                                  newCachedDocument:documentWrapper
+                                   deviceTimeToLive:deviceTimeToLive
+                                          operation:operation];
         completionHandler(documentWrapper);
       }
     }
   });
 }
 
+- (void)listDocumentsWithType:(Class)documentType
+                           partition:(NSString *)partition
+                         baseOptions:(MSBaseOptions *_Nullable)baseOptions
+                    cachedTokenBlock:(void (^)(MSCachedTokenCompletionHandler))cachedTokenBlock
+                 remoteDocumentBlock:(void (^)(MSPaginatedDocumentsCompletionHandler))remoteDocumentBlock
+                   completionHandler:(MSPaginatedDocumentsCompletionHandler)completionHandler {
+
+  // Retrieve a cached token.
+  cachedTokenBlock(^(MSTokensResponse *_Nullable tokensResponse, NSError *_Nullable error) {
+    // Handle error.
+    if (error) {
+      NSString *message =
+          [NSString stringWithFormat:@"Error while retrieving cached token, aborting operation: %@", [error localizedDescription]];
+      MSLogError([MSData logTag], @"%@", message);
+      MSDataError *dataError = [[MSDataError alloc] initWithErrorCode:MSACDataErrorCachedToken innerError:nil message:message];
+      MSPaginatedDocuments *documents = [[MSPaginatedDocuments alloc] initWithError:dataError
+                                                                          partition:partition
+                                                                       documentType:documentType];
+      completionHandler(documents);
+      return;
+    }
+
+    // Extract first token.
+    MSTokenResult *token = tokensResponse.tokens.firstObject;
+
+    // Retrieve from cache when offline and when there are pending operations
+    if (![self shouldAttemptRemoteOperationForPartition:partition]) {
+      MSPaginatedDocuments *cachedDocumentsList = [self.documentStore listWithToken:token
+                                                                          partition:partition
+                                                                       documentType:documentType
+                                                                        baseOptions:baseOptions];
+      if ([self.reachability currentReachabilityStatus] != NotReachable && [[cachedDocumentsList currentPage] items].count == 0) {
+        MSLogInfo([MSData logTag], @"Performing remote operation, since the local list is empty");
+        remoteDocumentBlock(^(MSPaginatedDocuments *_Nonnull remoteDocuments) {
+          // Update local store with the remote list of documents
```

Comment:
```suggestion

          // Update local store with the remote list of documents
```

Reply:
It's within a block, the formatter won't let this new line be because it's the first line of the block unfortunately.